sqlmesh/core/config/linter.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlglot import exp
+from sqlglot.helper import ensure_collection
+
+from sqlmesh.core.config.base import BaseConfig
+
+from sqlmesh.utils.pydantic import (
+    ValidationInfo,
+    field_validator,
+)
+
+
+class LinterConfig(BaseConfig):
+    """Configuration for model linting
+
+    Args:
+        enabled: Flag indicating whether the linter should run
+
+        rules: A list of error rules to be applied on model
+        warn_rules: A list of rules to be applied on models but produce warnings instead of raising errors.
+        exclude_rules: A list of rules to be excluded/ignored
+
+    """
+
+    enabled: bool = False
+
+    rules: t.Set[str] = set()
+    warn_rules: t.Set[str] = set()
+    exclude_rules: t.Set[str] = set()
+
+    @classmethod
+    def _validate_rules(cls, v: t.Any, data: t.Dict[str, t.Any]) -> t.Set[str]:
```

Comment:
It looks like we don't need `data`, both here and in L46 (`info`), L47 (`info.data`).